### ISSUE TITLE
fatal error throws when enable the --clean-up flags for kube proxy

### ISSUE THREAD

**Issue by hchenxa** [kubernetes/kubernetes#66819]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:
```
root@hchenxa-2:~# kubernetes/server/bin/kube-proxy  --kubeconfig=/var/lib/kubelet/kube-proxy-config --hostname-override=9.111.255.131 --cluster-cidr=10.1.0.0/16 --cleanup
W0731 09:46:25.248141   26014 server.go:195] WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.
I0731 09:46:25.268579   26014 server.go:448] Version: v1.11.1
F0731 09:46:25.335474   26014 server.go:361] <nil>
```
**What you expected to happen**:
The `kubeproxy` should be start up.

**How to reproduce it (as minimally and precisely as possible)**:
use k8s 1.11.1

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.11.1
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): ubuntu 16.04 LTS
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by corest** [comment]
Happened the same with 1.11

**Environment:**

Kubernetes version (use kubectl version): v1.11.1
Cloud provider or hardware configuration:
OS (e.g. from /etc/os-release): coreos 1745.7.0 
Kernel (e.g. uname -a): 4.14.48
Install tools:
Others:

**Comment by lewandom** [comment]
Seeing that too on both v1.13.0 and v1.13.4. Trying to clean up iptables rules after `kubeadm reset`:
```
$ sudo docker run -ti --rm --privileged --net=host -v /lib/modules:/lib/modules k8s.gcr.io/kube-proxy:v1.13.0 kube-proxy --cleanup
W0318 12:51:46.729140       1 server.go:194] WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.
I0318 12:51:46.771933       1 server.go:464] Version: v1.13.0
F0318 12:51:47.165929       1 server.go:377] <nil>
$ sudo docker run -ti --rm --privileged --net=host -v /lib/modules:/lib/modules k8s.gcr.io/kube-proxy:v1.13.4 kube-proxy --cleanup
W0318 12:52:01.669537       1 server.go:198] WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.
I0318 12:52:01.694629       1 server.go:483] Version: v1.13.4
F0318 12:52:02.066175       1 server.go:396] <nil>
```

I'm willing to debug it further, but don't know what to do as a next step.

**Comment by danwinship** [comment]
The problem is:

    	klog.Fatal(opts.Run())

in `NewProxyCommand` in `cmd/kube-proxy/app/server.go`.

#63532 rewrote kube-proxy to not use kubectl's utility functions but accidentally made it so that now it always logs the output of `opts.Run()` as a fatal error, rather than only doing so if the return value is non-`nil`. Should be an easy fix.

**Comment by freehan** [comment]
Fixed in 1.14.

**Comment by benceszikora** [comment]
@freehan We just got the same error on 1.15.0. It doesn't look to me like this was fixed: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server.go#L444
